### PR TITLE
feat: add quorum_up and quorum_down in virtual_server

### PR DESCRIPTION
**Pull request by rotulet**
add missing virtual_server option: quorum_up and quorum_down.
Tests have been corrected accordingly.

### CODE REVIEW

File: manifests/lvs/virtual_server.pp [hunk 1]
```diff
@@ -116,6 +124,8 @@
   Optional[Integer[1]] $persistence_timeout = undef,
   Enum['TCP','UDP'] $protocol = 'TCP',
   Optional[Integer[1]] $quorum = undef,
+  Optional[String[1]] $quorum_up = undef,
```
Comment (bastelfreak) [inline manifests/lvs/virtual_server.pp]: are those always paths to files? If so, you could use the `Stdlib::Absolutepath` datatype.

Comment (rotulet) [reply manifests/lvs/virtual_server.pp]: As I did not what kind of type to choose, so I selected the one used in [script.pp](https://github.com/voxpupuli/puppet-keepalived/blob/master/manifests/vrrp/script.pp). Is it correct?

Comment (Dan33l) [reply manifests/lvs/virtual_server.pp]: extract from man `keepalived.conf` : 
> Script to launch when quorum is gained. quorum_up

So data type `Stdlib::Absolutepath` looks correct.

Comment (alexjfisher) [reply manifests/lvs/virtual_server.pp]: > are those always paths to files?

Optionally followed by arguments...

File: manifests/lvs/virtual_server.pp [hunk 2]
```diff
@@ -116,6 +124,8 @@
   Optional[Integer[1]] $persistence_timeout = undef,
   Enum['TCP','UDP'] $protocol = 'TCP',
   Optional[Integer[1]] $quorum = undef,
+  Stdlib::Absolutepath $quorum_up = undef,
```
Comment (alexjfisher) [inline manifests/lvs/virtual_server.pp]: I think `String[1]` was correct here.  Otherwise you can't follow the path to the script with any arguments.

**Comment by rotulet** [comment]
Do we are agreed to use Optional[String[1]] as datatype for quorum_up/quorum_down? (I also made a mistake in the test corrections) 
Should I just revert my last commit?

File: manifests/lvs/virtual_server.pp (hunk 2 above)
Comment (rotulet) [reply manifests/lvs/virtual_server.pp]: Do we are agreed to use Optional[String[1]] as datatype for quorum_up/quorum_down? (I also made a mistake in the test corrections)
Should I just revert my last commit?

Comment (alexjfisher) [reply manifests/lvs/virtual_server.pp]: Or rebase to remove it.

Comment (rotulet) [reply manifests/lvs/virtual_server.pp]: done

**Comment by bastelfreak** [comment]
thanks @rotulet !

**Pull request merged**